renderdoc/driver/gl/wrappers/gl_emulated.cpp

```diff
@@ -2876,17 +2876,34 @@ void APIENTRY _glGetTexImage(GLenum target, GLint level, const GLenum format, co
       GL.glTexParameteri(target, eGL_DEPTH_STENCIL_TEXTURE_MODE, eGL_DEPTH_COMPONENT);
     }
 
-    // only support 2D textures for now
-    RDCASSERT(target == eGL_TEXTURE_2D, target);
+    // support 2D/Array/3D textures for now
+    RDCASSERT((target == eGL_TEXTURE_2D) || (target == eGL_TEXTURE_2D_ARRAY) ||
+                  (target == eGL_TEXTURE_3D),
+              target);
     GL.glGenTextures(1, &readtex);
     GL.glBindTexture(target, readtex);
 
     // allocate the texture
     GL.glTexParameteri(target, eGL_TEXTURE_MAX_LEVEL, 0);
-    GL.glTexImage2D(target, 0, internalformat, width, height, 0, remapformat, remaptype, NULL);
+    if(target == eGL_TEXTURE_2D)
+    {
+      GL.glTexImage2D(target, 0, internalformat, width, height, 0, remapformat, remaptype, NULL);
+    }
+    else
+    {
+      GL.glTexImage3D(target, 0, internalformat, width, height, depth, 0, remapformat, remaptype,
+                      NULL);
+    }
 
     // render to it
-    GL.glFramebufferTexture2D(eGL_FRAMEBUFFER, eGL_COLOR_ATTACHMENT0, target, readtex, 0);
+    if(target == eGL_TEXTURE_2D)
+    {
+      GL.glFramebufferTexture2D(eGL_FRAMEBUFFER, eGL_COLOR_ATTACHMENT0, target, readtex, 0);
+    }
+    else
+    {
+      GL.glFramebufferTextureLayer(eGL_FRAMEBUFFER, eGL_COLOR_ATTACHMENT0, readtex, 0, 0);
+    }
```

Comment:
I think it would be clearer to merge these ifs together so the reader has less to keep track of, since the branches are the same.

Reply:
That's not quite what I meant, I was suggesting to merge the two ifs since the condition was the same in both. `glFramebufferTextureLayer` is not valid to call on 2D textures so this change will no longer work for the plain 2D texture case.

Reply:
Thank you. I got your message. Resolved.